core/src/main/java/org/infinispan/distribution/DistributionInfo.java

```diff
@@ -0,0 +1,66 @@
+package org.infinispan.distribution;
+
+import java.util.List;
+
+import org.infinispan.distribution.ch.ConsistentHash;
+import org.infinispan.remoting.transport.Address;
+
+public class DistributionInfo {
```

Comment:
at least, it can be simplified. You just need the list of owners (first owner in the list is the primary) and an enum to say if it is the primary/backup/non-owner :)
It would allow to u￼ se a switch statement instead of using a bunch of if...
this is my opinion of course :)